### PR TITLE
Fix CVE-2025-30065 and CVE-2025-24970

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Bump `parquet-avro` to `1.15.1` and `netty-all` to `4.8.118.Final`

## Brief change log

- Fix critical and high CVE

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -372,6 +373,12 @@
                 <version>${parquet-avro.version}</version>
             </dependency>
 
+            <dependency>
```
Comment (zhoujinsong) [inline pom.xml:376]: It seems `parquet-avro` is dependent on `parquet-jackson` in runtime scope after pumping up to 1.15.1; this is the reason why we need it here.

File: amoro-format-iceberg/pom.xml [hunk 2]
```diff
@@ -134,6 +134,11 @@
             <artifactId>parquet-avro</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (zhoujinsong) [inline amoro-format-iceberg/pom.xml:137]: Do we need to declare it a dependency here, as `amoro-format-iceberg` has already dependent on `amoro-common`?

File: amoro-format-mixed/amoro-mixed-trino/src/test/java/org/apache/amoro/trino/mixed/TestKeyedTable.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public void testStats() {
         .matches(
             "VALUES "
                 + "('id', NULL, NULL, 0e0, NULL, '1', '4'), "
-                + "('name$name', 429e0, NULL, 0e0, NULL, NULL, NULL), "
+                + "('name$name', 461e0, NULL, 0e0, NULL, NULL, NULL), "
```
Comment (zhoujinsong) [inline amoro-format-mixed/amoro-mixed-trino/src/test/java/org/apache/amoro/trino/mixed/TestKeyedTable.java:65]: I have no idea why this change is needed.

**Review by zhoujinsong — COMMENTED** [review]
@XBaith Thanks for driving this, I left some comments, PTAL.

File: pom.xml (hunk 1 above)
Comment (XBaith) [reply pom.xml:376]: Actually, parquet-avro 1.15.1 contains parquet-jackson 1.15.1 which lead fail to complie the project with JDK8
I replace paruqet-jackson 1.15.1 with 1.13.1 to build the project correctly.

File: amoro-format-iceberg/pom.xml (hunk 2 above)
Comment (XBaith) [reply amoro-format-iceberg/pom.xml:137]: +1. the parquet-jackson and parquet-avro are included

File: pom.xml (hunk 1 above)
Comment (zhoujinsong) [reply pom.xml:376]: Got it, we may remove this after upgrading the JDK version, maybe add a comment to add this TODO.

File: amoro-format-iceberg/pom.xml (hunk 2 above)
Comment (zhoujinsong) [reply amoro-format-iceberg/pom.xml:137]: Yes, so we may drop the dependencies here?

File: amoro-format-mixed/amoro-mixed-trino/src/test/java/org/apache/amoro/trino/mixed/TestKeyedTable.java (hunk 3 above)
Comment (XBaith) [reply amoro-format-mixed/amoro-mixed-trino/src/test/java/org/apache/amoro/trino/mixed/TestKeyedTable.java:65]: `data_size` is an estimate and I'm not sure why it changes. I'm guessing it's the encoding strategy that causes the file size to change.
